lib/schemas/rfc7239.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+function splitMap (string, separator, cb) {
+  // split into elements
+  return string.split(separator)
+    .filter(Boolean)
+    .forEach(cb)
+}
+
+function parsePart (part) {
+  var pair = part.split(/ *= */)
+
+  var name = pair[0].toLowerCase()
+  var value = pair[1]
+
+  if (value) {
+    switch (typeof this[name]) {
+      case 'undefined':
+        this[name] = value
+        break
+
+      // convert to array
+      case 'string':
+        this[name] = [this[name], value]
+        break
+
+      // append to array
+      case 'object':
+        this[name].push(value)
+        break
+    }
+  }
+}
+
+var ELEMENT_SEPARATOR = / *; */
+var PART_SEPARATOR = / *, */
+
+module.exports = function (req) {
+  var forwarded = {}
+  var header = req.headers.forwarded
+
+  if (!header) {
+    return forwarded
+  }
+
+  splitMap(header, ELEMENT_SEPARATOR, function parseElement (el) {
```

Comment:
>  The current implementation will do the wrong thing

can you clarify what you mean by that?

the [token syntax](http://tools.ietf.org/html/rfc7230#section-3.2.6) does not actually allow a comma to be present: 

```
     token          = 1*tchar

     tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
                    / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
                    / DIGIT / ALPHA
                    ; any VCHAR, except delimiters
```

```
Delimiters are chosen from the set of US-ASCII visual characters not allowed in a token
(DQUOTE and "(),/:;<=>?@[\]{}").
```
**so, if anything, we should discard `for="foo,bar"` since its not a valid token**

###### verification vs. validation

further, imo, the purpose of this library is to *parse* the headers, ***not to verify values***, the resulting values should be filtered through a secondary layer if the implementer chooses to verify values, such as network addresses, host names, etc ...

which leaves us with the question: **should we add validation here?**

specifically: to validate *addresses*, *hostnames*, *ports*.

---

as a side, RFC 7239 node identifier syntax could also include [unknown](http://tools.ietf.org/html/rfc7239#section-6.2) and [obfuscated](http://tools.ietf.org/html/rfc7239#section-6.3) identifiers.